### PR TITLE
Added dartfmt fixer

### PR DESCRIPTION
**Pull request by reisub0**
Simply adapted the code from gofmt. Found these by grepping for gofmt and creating similar files but with dartfmt. I might have missed something but WFM.

Signed-off-by: Govind <k.p.govind@gmail.com>

### CODE REVIEW

File: doc/ale-dart.txt [hunk 1]
```diff
@@ -35,4 +35,15 @@ g:ale_dart_dartanalyzer_executable         *g:ale_dart_dartanalyzer_executable*
 
 
 ===============================================================================
+dartfmt                                                       *ale-dart-dartfmt*
+
+g:ale_dart_dartfmt_option                           *g:ale_dart_dartfmt_options*
+                                                   *b:ale_dart_dartfmt_options*
```
Comment (w0rp) [inline doc/ale-dart.txt]: Align tags to the right margin.

File: doc/ale-dart.txt [hunk 2]
```diff
@@ -35,4 +35,15 @@ g:ale_dart_dartanalyzer_executable         *g:ale_dart_dartanalyzer_executable*
 
 
 ===============================================================================
+dartfmt                                                       *ale-dart-dartfmt*
+
+g:ale_dart_dartfmt_option                           *g:ale_dart_dartfmt_options*
```
Comment (w0rp) [inline doc/ale-dart.txt]: Don't forget to document the executable option too.

**Comment by reisub0** [comment]
Can you please tell me what I need to do fix the last test on Travis (mismatched ToC, Headings)

**Comment by w0rp** [comment]
Yes, that's telling you that you need to add `dartfmt` to the table of contents in `doc/ale.txt` with a link which matches the tag for the heading.

**Comment by reisub0** [comment]
> ========================================
Look for badly aligned doc tags
========================================
Badly aligned tags follow:
========================================
Look for table of contents issues
========================================
Check for bad ToC sorting:
Check for mismatched ToC and headings:
Something went wrong!
The command "./run-tests
" exited with 1.
Done. Your build exited with 1.

Still seems to be failing. On my vim, the doc Tags are working just fine and I can navigate between the main page and the corresponding entry in the ale-dart file

**Comment by reisub0** [comment]
All fixed!

**Comment by w0rp** [comment]
Cheers! :beers:

**Pull request merged**